browser/brave_browser_process_impl.cc

```diff
@@ -344,10 +399,14 @@ brave::BraveP3AService* BraveBrowserProcessImpl::brave_p3a_service() {
 
 brave::BraveReferralsService*
 BraveBrowserProcessImpl::brave_referrals_service() {
-  if (!brave_referrals_service_)
+  if (!brave_referrals_service_) {
     brave_referrals_service_ = std::make_unique<brave::BraveReferralsService>(
         local_state(), brave_stats::GetAPIKey(),
         brave_stats::GetPlatformIdentifier());
+    brave_referrals_service_->set_delegate(
+        std::make_unique<ReferralServiceDelegate>(
+            brave_referrals_service_.get()));
```

Comment:
According to the comment (https://source.chromium.org/chromium/chromium/src/+/main:base/memory/raw_ptr.h;l=1456?q=raw_ptr.h), using `get()` should be avoided.

Reply:
and how about inserting `brave_referrals_service()` decls and defs into `ENABLE_BRAVE_REFERRALS` guard?

Reply:
> get() should be avoided.

Actually, brave_referral_service_ is a unique_ptr<> so I think it's okay to leave as is.

Reply:
Oops, I was confusing with `ReferralServiceDelegate::service_` :)